File: apps/server-comet/src/rabbitmq/publisher.ts
import { getConnection } from './config';
import { ChatMessage } from './types/message';

export async function sendMessage(message: ChatMessage) {
	try {
		const conn = await getConnection();
		const channel = await conn.createChannel();
		const buffer = Buffer.from(JSON.stringify(message));
		channel.assertExchange('chat', 'fanout', { durable: false });
		channel.publish('chat', '', buffer);
		console.log('Sending to queue from comet');
	} catch (error) {
		console.log('[Comet] Error while sending to queue: ', error);
	}
}

File: apps/server-orbit/app/rabbitmq/publisher.py
import json
from .config import setup_queue
from .schemas import ChatMessage

def send_message(message: ChatMessage):
    channel = setup_queue()[1]
    body = json.dumps(message.model_dump())
    channel.exchange_declare(exchange='chat', exchange_type='fanout', durable=False) # type: ignore
    channel.basic_publish(exchange='chat', routing_key='', body=body)
    print("[X] Orbit sent: ", body)